### DISCUSSION TITLE
HistFactory staterror functionality in pyhf

### DISCUSSION THREAD

**Discussion by iokarkan** [scikit-hep/pyhf#1394] [Q&A]
In an effort to replicate fit results from HistFactory workspace using pyhf, I've come across the staterror configurations of HistFactory, as seen in [https://root.cern/doc/v606/classRooStats_1_1HistFactory_1_1StatErrorConfig.html](url).

For example, I am trying to understand whether the staterror of pyhf can accommodate the 
```channel.SetStatErrorConfig(0.001, "Poisson")``` 
functionality of HistFactory, with the first number being an uncertainty threshold (per bin, I imagine) and the second being the constraint type, where "Gaussian" or "Poisson" is selectable. From pyhf's documentation, the constraint type of staterror is a product of Gaussians, and I cannot find anywhere indicating that this is optional.

I have tried making an xml workspace using the pyhf CLI, and adding the 
```  <StatErrorConfig RelErrorThreshold="0.5" ConstraintType="Poisson" />```
where necessary, and trying to go back to a .json workspace, again with the CLI, but I am not noticing any changes.

Is this functionality to be expected in the future, or is it a design choice for pyhf? Thanks!

**Comment by alexander-held** [comment]
The type of constraint term can currently not be switched to Poisson, see https://github.com/scikit-hep/pyhf/issues/760. The threshold setting is also not respected at the moment, see https://github.com/scikit-hep/pyhf/issues/662.

**Reply by iokarkan** [reply]
Apologies for not finding these in the pyhf Issues! I tried searching the code but I must have missed the issue reports!

**Reply by iokarkan** [reply]
Would manually setting the staterror "data" to be 0, during the creation of the pyhf workspace, for a bin that doesn't surpass the set threshold be a work-around for the threshold part of the question?

**Reply by alexander-held** [reply]
Setting the value to `0.0` will cause another issue (https://github.com/scikit-hep/pyhf/issues/293#issuecomment-734885749), but you can use some small value like `0.000001` as a workaround.
